### PR TITLE
estimate fees and readme fixes

### PR DESCRIPTION
**Pull request by domwoe**
Added your logic for fee estimation from the donation example and updated the readme and some comments a bit.
Furthemore, the script to create only one event.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,12 +173,12 @@ Here you can find a number of examples leveraging the chain_fusion starter logic
...
--   A ckNFT canister that mints an NFT on the ICP when an EVM helper smart contract emits an `ReceivedNft`, similar to the [`EthDepositHelper`](https://github.com/dfinity/ic/blob/master/rs/ethereum/cketh/minter/EthDepositHelper.sol) contract the ckETH minter uses. This could enable users to trade NFTs on the ICP without having to pay gas fees on Ethereum.
+-   A ckNFT canister that mints an NFT on the ICP when an EVM helper smart contract emits a `ReceivedNft`, similar to the [`EthDepositHelper`](https://github.com/dfinity/ic/blob/master/rs/ethereum/cketh/minter/EthDepositHelper.sol) contract the ckETH minter uses. This could enable users to trade NFTs on the ICP without having to pay gas fees on Ethereum.
 -   Price oracles for DeFi applications via [exchange rate canister](https://github.com/dfinity/exchange-rate-canister)
 -   Prediction market resolution
 -   Soulbound NFT metadata and assets stored in a canister
--   An on-chain managed passive index fonds (e.g. top 10 ERC20 tokens trade on Uniswap)
+-   An on-chain managed passive index fund (e.g. top 10 ERC20 tokens trade on Uniswap)
```
Comment (letmejustputthishere) [inline README.md:183]:
```suggestion
-   An on-chain managed passive index fund (e.g. top 10 ERC20 tokens traded on Uniswap)
```

File: script/Coprocessor.s.sol [hunk 2]
```diff
@@ -13,16 +13,13 @@ contract MyScript is Script {
...
-        // first need to create the canister, deploy the NFT contract passing the canister url as baseURI
-        // and then deploying the canister passing the NFT contract address as a deploy argument.
+
         Coprocessor coprocessor = new Coprocessor(chain_fusion_canister_address);
 
         // we can call the mint function to mint a token
         // the address we mint to belongs to the deployerPrivateKey
         // again, the transaction is signed with the deployerPrivateKey
-        for (uint256 index = 0; index < 10; index++) {
+        for (uint256 index = 0; index < 1; index++) {
```
Comment (letmejustputthishere) [inline script/Coprocessor.s.sol:22]: why not run this 10 times?

File: src/chain_fusion_backend/src/job/submit_result.rs [hunk 3]
```diff
@@ -12,22 +13,23 @@ pub async fn submit_result(result: String, job_id: U256) {
...
 
-    // let mut data = keccak256(function_signature).as_ref()[0..4].to_vec();
-    // data.extend(ethers_core::abi::AbiEncode::encode(result));
     let selector = &keccak256(function_signature.as_bytes())[0..4];
     let args = (result, job_id).encode();
     let mut data = Vec::from(selector);
     data.extend(args);
 
+    let gas_limit = U256::from(5000000);
+    let fee_estimates = fees::estimate_transaction_fees(9).await;
```
Comment (letmejustputthishere) [inline src/chain_fusion_backend/src/job/submit_result.rs:22]: should we pas those to the function like we do for `transfer_eth`?

**Review by letmejustputthishere — CHANGES_REQUESTED** [review]

File: src/chain_fusion_backend/src/job/submit_result.rs (hunk 3 above)
Comment (domwoe) [reply src/chain_fusion_backend/src/job/submit_result.rs:22]: Which function?

Refactor to `call_contract(sig, data, gas_limit)` or something?

File: script/Coprocessor.s.sol (hunk 2 above)
Comment (domwoe) [reply script/Coprocessor.s.sol:22]: Too many logs in the console :)

Comment (letmejustputthishere) [reply script/Coprocessor.s.sol:22]: lets find a middle ground then :D If you really only want to run it once we can get rid of the entire loop

File: src/chain_fusion_backend/src/job/submit_result.rs (hunk 3 above)
Comment (letmejustputthishere) [reply src/chain_fusion_backend/src/job/submit_result.rs:22]: i meant `submit_result`. 
within both `transfer_eth` and `submit_result` we 
1. create a request
2. sign it
3. send it to the network
4. check the status 

for `transfer_eth` we pass the `gas` and `fee_estimate` to the function, for `submit_result` it's hardcoded in the body of the function. ideally both behave the same. i think the `fee_estimate` should be part of the body in both cases, so we only need to decide on how to proceed with `gas`

**Pull request merged**